events/loginEvent.py

```diff
@@ -90,6 +147,26 @@ async def handle(web_handler: AsyncRequestHandler) -> tuple[str, bytes]:  # toke
             # Invalid password
             raise exceptions.loginFailedException()
 
+        try:
+            # we have a user ID we can rely on, allow further processing of login body
+            login_data = parse_login_data(web_handler.request.body)
+
+            amqp_login_message = login_data | {"user_id": userID}
+
+            # don't transport the `password_md5` key
+            del amqp_login_message["password_md5"]
+
+            for routing_key in settings.BANCHO_LOGIN_ROUTING_KEYS:
+                await glob.amqp_channel.default_exchange.publish(
+                    aio_pika.Message(body=orjson.dumps(amqp_login_message)),
+                    routing_key=routing_key,
+                )
+        except Exception:  # don't allow publish failure to block login
+            logger.warning(
+                "[Non-blocking] Failed to send bancho login request through AMQP",
+                exc_info=True,
```

Comment:
nit for extra data in this case
```suggestion
                exc_info=True,
                extra={"user_id": userID},
```